lib/endpoints/class-wp-json-posts-controller.php

```diff
@@ -1124,6 +1124,15 @@ protected function prepare_links( $post ) {
 			);
 		}
 
+		if ( 'attachment' !== $post->post_type ) {
```

Comment:
> Core doesn't have an explicit way to note if a post supports attachments.

I've opened a new issue so it doesn't block this PR #1029 

> A non-explict option would be to check if a post_type supports either thumbnail OR editor.

Seems a bit prone to breakage. Let's instead also exclude `nav_menu_item` and `revision`, and I think that'll be fine for now.